### PR TITLE
Update README to better address some of the problems I had getting started

### PR DESCRIPTION
**Pull request by carlbennettnz**
Specifically:
- Fix first sentence syntax
- Fix typo in example
- Add note that JSON API collection names are also dasherized
- Add some detail to the documentation of the `beforeRender` hook
- Add a section on filtering. I don't really know much about how it
  works, so it's a little sparse, but it's something.

This is my first ever pull request (woohoo :tada:) so please let me know if there's anything I should be doing differently :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,7 +35,7 @@ Check out the [full, working example repo](http://github.com/ethanresnick/json-a
   });
 
   registry.type("places", {
-    adapter: adapter,
+    dbAdapter: adapter,
```
Comment (ethanresnick) [inline README.md]: Thanks!

File: README.md [hunk 2]
```diff
@@ -66,14 +66,14 @@ Check out the [full, working example repo](http://github.com/ethanresnick/json-a
 
 # Core Concepts
 ## Resource Type Descriptions <a name="resource-type-descriptions"></a>
-The JSON-API spec is built around the idea of typed resource collections. For example, you can have a `"people"` collection and a `"companies"` collection. (By convention, type names are plural and lowercase.)
+The JSON-API spec is built around the idea of typed resource collections. For example, you can have a `"people"` collection and a `"companies"` collection. (By convention, type names are plural, lowercase, and dasherized.)
```
Comment (ethanresnick) [inline README.md]: :+1:

File: README.md [hunk 3]
```diff
@@ -66,14 +66,14 @@ Check out the [full, working example repo](http://github.com/ethanresnick/json-a
...
 - `dbAdapter`: the [database adapter](#database-adapters) to use to find and update these resources. By specifying this for each resource type, different resource types can live in different kinds of databases.
 
-- <a name="before-render"></a>`beforeRender` (optional): a function called on each resource after it's found by the adapter but before it's sent to the client. This lets you do things like hide fields that some users aren't authorized to see.
+- <a name="before-render"></a>`beforeRender` (optional): a function called on each resource after it's found by the adapter but before it's sent to the client. This lets you do things like hide fields that some users aren't authorized to see. It receives `resource`, `req`, `res`, and `superFn` as input and either a transformed resource or a promise as output. Returning or resolving to `undefined` will remove the resource from the response.
```
Comment (ethanresnick) [inline README.md]: This is very good info to document, but it seems a bit out of place in the readme, which is about describing the library's overall capabilities more than documenting specific arguments.

Maybe you could create a `beforeRender` function in one of the resource descriptions in the example repo, and add a [jsdoc block]() on top of that `beforeRender` explaining all the arguments/return values. Then, this readme could just include a link to the line with that docbloc?

Comment (carlbennettnz) [inline README.md]: Sounds good to me :+1: 

To revert this change do I need to push another commit to my fork, or are you able to merge just part of the pull request? Sorry, new to this whole contributing thing

File: README.md [hunk 4]
```diff
@@ -83,6 +83,9 @@ To use this library, you describe the special behavior (if any) that resources o
 
 -  <a name="info"></a>`info` (optional): this allows you to provide extra information about the resource that will be included in the documentation. Available properties are `"description"` (a string describing what resources of this type are) and `"fields"`. `"fields"` holds an object in which you can describe each field in the resource (e.g. listing validation rules). See the [example implemenation](https://github.com/ethanresnick/json-api-example/blob/master/src/resource-descriptions/schools.js) for more details.
 
+## Filtering
```
Comment (ethanresnick) [inline README.md]: Thanks! This whole section was much needed, and it looks great. 

One thing, though, is that I've been a bit hesitant to document the support for Mongo query operators, because this support may be removed in future versions of the library (because such operators could open up security vulnerabilities around [nosql injection](https://www.owasp.org/index.php/Testing_for_NoSQL_injection) and their syntax is database-specific, whereas the final built-in filter set would ideally be backend agnostic). I'm fine mentioning them, though, as long as there's also a sentence or two warning that they might be removed. Do you think you could add that?

Comment (carlbennettnz) [inline README.md]: Sure thing

File: README.md (hunk 3 above)
Comment (ethanresnick) [inline README.md]: No worries! I can clone your fork, cherry-pick your commit, and then edit it before merging, but the easiest thing is for you to push a new commit. You can even use rebase (since nothing depends on your fork yet) to merge your new commit into your last one, to make the history a bit neater, and then force-push that new version.

Something like:

1. `git commit -m "message here doesn't matter"`
2. `git rebase -i HEAD~2`, which'll select the last two commits for rebase, and then you can squash the second into the first and write a new commit message (or use the first commit's message) for the new commit.
3. `git push --force`

(Apologies if any of the above is remedial, or bad practice—git's intricacies are a bit of a mystery to me—but the above is what I usually do when making a PR, and it's never caused problems.)

**Comment by carlbennettnz** [comment]
How's that?

**Comment by ethanresnick** [comment]
I'll merge this now, and we can address beforeRender in a separate PR. Thanks again :)

**Comment by carlbennettnz** [comment]
No problem :)
